gender_novels/corpus.py

```diff
@@ -64,25 +99,57 @@ def count_authors_by_gender(self, gender):
 
         :rtype: int
         """
+        c_copy = self.filter_by_gender(gender)
+        return len(c_copy.novels)
+
+    def filter_by_gender(self, gender):
+        """
+        Return a new Corpus object that contains only authors whose gender
+        matches gender_filter.
+
+        Accepted inputs are 'male', 'female', 'non-binary' and 'unknown'
+        but no abbreviations.
+
+        >>> from gender_novels.corpus import Corpus
+        >>> c = Corpus('sample_novels')
+        >>> female_corpus = c.filter_by_gender('female')
+        >>> len(female_corpus.novels)
+        38
+        >>> female_corpus.novels[0].title
+        'Little Women'
+
+        >>> male_corpus = c.filter_by_gender('male')
+        >>> len(male_corpus.novels)
+        60
+
+        >>> male_corpus.novels[0].title
+        'Heart of Darkness'
+
+        :param gender: gender name
+        :return: Corpus
+        """
         supported_genders = ('male', 'female', 'non-binary', 'unknown')
         if gender not in supported_genders:
             raise ValueError(
                 f'Gender must be {", ".join(supported_genders)} '
                 + f'but not {gender}.')
 
-        # check if all novels have an author_gender attribute
+        c_copy = self.clone()
```

Comment:
Can you be more explicit in the `c_copy` variable?

Reply:
right on it!